### PR TITLE
Dependent Fields: Remove mock call for `expense_fields` and add unit tests

### PR DESCRIPTION
**Pull request by yksolanki9**
- Also removed unused keys in expense forms and added types

Code Coverage:
Screenshot 2023-02-20 at 5 20 11 PM
Screenshot 2023-02-20 at 4 34 50 PM

### CODE REVIEW

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 1]
```diff
@@ -2284,9 +2284,7 @@ export class AddEditPerDiemPage implements OnInit {
             .pipe(
               //TODO: Remove the delay once APIs are available
```
Comment (Julias0) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:2285]: remove this too

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 2]
```diff
@@ -2310,7 +2308,6 @@ export class AddEditPerDiemPage implements OnInit {
     this.dependentFields.push({
       id: dependentField.id,
       parentFieldId: dependentField.parent_field_id,
-      parentFieldValue: dependentField.parent_field_value,
```
Comment (Julias0) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:2313]: is this intentional?

File: src/app/core/services/custom-inputs.service.ts [hunk 3]
```diff
@@ -29,8 +29,6 @@ export class CustomInputsService {
   @Cacheable({
     cacheBusterObserver: customInputssCacheBuster$,
   })
-  //TODO: Remove this mapping and fix type once APIs are available
-  //These are for 'Yash's Test Organization' on staging
   getAll(active: boolean): Observable<any[]> {
```
Comment (Julias0) [inline src/app/core/services/custom-inputs.service.ts:32]: Add the data type here - this has the model type of expenseField

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts (hunk 2 above)
Comment (yksolanki9) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:2313]: Yes, we're not using it in any way. I had added it for another approach, but missed removing it

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts (hunk 1 above)
Comment (yksolanki9) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:2285]: Done

File: src/app/core/services/custom-inputs.service.ts (hunk 3 above)
Comment (yksolanki9) [reply src/app/core/services/custom-inputs.service.ts]: Done

File: src/app/core/test-data/custom-inputs.spec.data.ts [hunk 4]
```diff
@@ -112,6 +112,63 @@ export const platformApiResponse = {
...
+      id: 218227,
+      code: null,
+      column_name: 'text_column8',
+      created_at: '2023-02-01T07:08:06.693990+00:00',
+      default_value: null,
+      field_name: 'CF1',
+      is_custom: true,
+      is_enabled: true,
+      is_mandatory: true,
+      options: ['CF1.1', 'CF1.2', 'CF1.3', 'CF1.4', 'CF1.5'],
```
Comment (Julias0) [inline src/app/core/test-data/custom-inputs.spec.data.ts:125]: For dependant select type inputs, the options wont come in the API response right?

Comment (yksolanki9) [reply src/app/core/test-data/custom-inputs.spec.data.ts]: Removed it, and made it optional in the model

**Review by Julias0 — APPROVED** [review]

**Pull request merged**